solutions/gold/usaco-694.mdx

```diff
@@ -165,5 +165,59 @@ public class HoofPaperScissors {
 }
 ```
 </JavaSection>
+<PySection>
+
+<Warning>
+
+Due to Python's constant factor, the below solution TLEs on a couple test cases.
+
+</Warning>
+
+```py
+with open("hps.in") as read:
+	n, k = map(int, read.readline().strip().split())
+	A = [0] * n
+	for i in range(n):
+		a = read.readline().strip()
+		if a == "H":
+			A[i] = 0
+		elif a == "P":
+			A[i] = 1
+		else:
+			A[i] = 2
+
+"""
+dp[i][j][k] is the largest number of games she wins at
+games i with switches j with current item k
+0 == H 1 == P 2 == S
+"""
+dp = [[[0] * 3 for _ in range(k + 2)] for _ in range(n + 1)]
```

Comment:
k+2? i've seen +1 in dp arrays but never +2

Reply:
Yes, since I am avoiding the magic numbers from the cpp version and since seeing the loops it is what is required.

Reply:
i mean it seems this +2 could be avoided by doing some rudimentary bounds checking